api/apps/geoprocessing/src/modules/scenario-planning-units-inclusion/scenario-planning-units-inclusion-processor.ts

```diff
@@ -59,10 +59,11 @@ export class ScenarioPlanningUnitsInclusionProcessor
       );
     }
 
-    if (!includeGeo && !excludeGeo) {
-      puIdsToInclude.push(...(job.data.include?.pu ?? []));
-      puIdsToExclude.push(...(job.data.exclude?.pu ?? []));
-    }
+    // After setting inclusions and exclusions from geometries above, add
+    // inclusions and exclusions by id, so that we effectively use the union
+    // of the two methods (byId and byGeoJson).
+    puIdsToInclude.push(...(job.data.include?.pu ?? []));
+    puIdsToExclude.push(...(job.data.exclude?.pu ?? []));
```

Comment:
for minor performance reasons, can get `unique` after both assignments have been made.

LGTM ✅

Reply:
👍🏼 3c0623995359f171675327f26870ac9d199e7bfa